src/main/java/duke/Event.java

```diff
@@ -0,0 +1,173 @@
+                System.out.println(e.toString());
+                System.out.println(divider);
+            }
+        } else {
+            // for(int j=1; j< splitInput.length; j++){
+            //     if(splitInput[j].equals("/from")){
+            //         for (int k=1; k< j-1; k++){
+            //             splitInput[1] = splitInput[1] + " " + splitInput[k+1];
+            //         }
+            //         for(int i = j + 1; i < splitInput.length; i++) {
+            //             splitInput[i] = splitInput[i].replace("/to", "to:");
+            //         }
+            //         for (int l=splitInput.length-1; l > j +1; l--){
+            //             splitInput[splitInput.length-1] = splitInput[l-1]+" "+splitInput[splitInput.length-1];
+            //         }
+            //     } else {
+            //         splitInput[j] = splitInput[j];
+            //     }
+            // }
+               
+            // String duration = splitInput[splitInput.length-1];
+            // String desc = splitInput[1];
+           
+           
+            String event = combinedString.split(" ", 2)[1];
+            String[] event_Arr = event.split(" /from", 2);
+
+                if (event_Arr.length == 2) {
+                    String desc = event_Arr[0];
+                    String[] period_Arr = event_Arr[1].split(" /to");
+                    if (period_Arr.length == 2) {
+
+                        String from = period_Arr[0].strip();
+                        String to = period_Arr[1].strip();;
+
+                        if(isDate(from) && isDate(to)){
+                            DateTimeFormatter formatter = DateTimeFormatter.ofPattern("yyyy-MM-dd");
+                            LocalDate ldFrom = LocalDate.parse(from, formatter);
+                            LocalDate ldTo = LocalDate.parse(to, formatter);
+                            Event e = new Event(desc, ldFrom, ldTo);
+                            array.add(e);
+                            System.out.println(divider);
+                            System.out.println("     Got it. I've added this task:");
+                            System.out.println("     " + e.toString());
+                            System.out.println("     Now you have " + array.size() + " tasks in the list.");
+                            System.out.println(divider);
+                        } else {
+                            Event e = new Event(desc, from, to);
+                            array.add(e);
+                            System.out.println(divider);
+                            System.out.println("     Got it. I've added this task:");
+                            System.out.println("     " + e.toString());
+                            System.out.println("     Now you have " + array.size() + " tasks in the list.");
+                            System.out.println(divider);
+                        }
+
+                       
+                    }
+                }
```

Comment:
Perhaps, this make use of [guard clauses] to reduce the indentations of the `if` statements, as per the ["make-the-happy-path-prominent"][link] and ["Avoid Deep Nesting"][nesting] coding guidelines, like so:





[guard clauses]: http://wiki.c2.com/?GuardClause
[link]: https://nus-cs2103-ay2223s2.github.io/website/se-book-adapted/chapters/codeQuality.html#make-the-happy-path-prominent
[nesting]: https://nus-cs2103-ay2223s2.github.io/website/se-book-adapted/chapters/codeQuality.html#avoid-deep-nesting

```java
if (event_Arr.length != 2) {
    return;
}

String desc = event_Arr[0];
String[] period_Arr = event_Arr[1].split(" /to");
if (period_Arr.length != 2) {
    return;
}

String from = period_Arr[0].strip();
String to = period_Arr[1].strip();;

if(isDate(from) && isDate(to)){
    DateTimeFormatter formatter = DateTimeFormatter.ofPattern("yyyy-MM-dd");
    LocalDate ldFrom = LocalDate.parse(from, formatter);
    LocalDate ldTo = LocalDate.parse(to, formatter);
    Event e = new Event(desc, ldFrom, ldTo);
    array.add(e);
    System.out.println(divider);
    System.out.println("     Got it. I've added this task:");
    System.out.println("     " + e.toString());
    System.out.println("     Now you have " + array.size() + " tasks in the list.");
    System.out.println(divider);
} else {
    Event e = new Event(desc, from, to);
    array.add(e);
    System.out.println(divider);
    System.out.println("     Got it. I've added this task:");
    System.out.println("     " + e.toString());
    System.out.println("     Now you have " + array.size() + " tasks in the list.");
    System.out.println(divider);
}
```